### ISSUE TITLE
(cmake) Not in Path

### ISSUE THREAD

**Issue by nazar-pc** [chocolatey-community/chocolatey-packages#2442]
### Checklist

- [X] I have verified that this is the correct repository, and the package is maintained by the chocolatey-community user.
- [X] I have verified that this is happening in the latest available version of the package.

### Chocolatey Version

2.2.2

### Chocolatey License

None

### Package Version

3.28.3

### Current Behaviour

CMake was the only package out of many that I installed that wasn't added to Path automatically by default. I have found old https://github.com/chocolatey-community/chocolatey-packages/issues/1444, but looks like back then it worked correctly.

### Expected Behaviour

CMake is added to Path on installation

### Steps To Reproduce

1. `choco install cmake`
2. `cmake` in newly open PowerShell window

### Environment

```markdown
- Operating System: Windows 11 23H2 Microsoft Windows NT 10.0.22631.0
- PowerShell Version: 5.1.22621.2506
- Shell: PowerShell
```

### Chocolatey Log

```shell
https://gist.github.com/nazar-pc/ede66d3fe2c366a48c306c3636a51123
```

**Comment by stenolog** [comment]
Confirming, just ran into this.

Fix it by manually updating the path variable

### Chocolatey Project Process

Wondering why the choco project closes issues that casually, degrading choco's quality:

- https://github.com/chocolatey-community/chocolatey-packages/issues/987#event-1526406970

**Comment by stenolog** [comment]
@AdmiringWorm, issues do not disappear by writing some comments, closing and locking issues.

**Comment by pauby** [comment]
@stenolog This is a community maintained repository. Packages are maintained by volunteers and anything done is on a best effort basis. If you need an issue fixed, then it is expected that you jump in and submit a PR. Not post sarcastic comments about something from 5 years ago. 

I've removed your comment above as it's unneeded and contributes nothing to the issue. Please refer to our [Code of Conduct](https://github.com/chocolatey-community/.github/blob/d77df364a2449acaef47580ccb7067496042d780/CODE_OF_CONDUCT.md) before posting further.

**Comment by pauby** [comment]
@nazar-pc as there was a [previous issue for this](https://github.com/chocolatey-community/chocolatey-packages/issues/987). It was made clear there that [the package is following the default behaviour of the installer (which does not add `cmake` to the path)](https://github.com/chocolatey-community/chocolatey-packages/issues/987#issuecomment-693282520). As this is default behaviour, this will be expected behaviour by those who have used that same installer outside Chocolatey CLI.

@AdmiringWorm, would a PR, adding a package parameter and making it clear in the output that the software is not on the path because of this default behaviour, be an option here?

**Comment by nazar-pc** [comment]
Even if it is the default upstream, I would argue is the wrong default. Anyone who installs CMake likely wants to use it in PowerShell or some other tooling and in most of those cases `cmake.exe` is expected to be in `PATH`. If it is not there, it just makes software harder to use.

I basically expect both `apt-get install cmake` and `choco install cmake` have about the same result and currently it is not the case. Especially so because I installed other dev tooling like `git` and `make` and they all were added to `PATH` just like I expected.

**Comment by pauby** [comment]
> Even if it is the default upstream, I would argue is the wrong default. Anyone who installs CMake likely wants to use it in PowerShell or some other tooling and in most of those cases cmake.exe is expected to be in PATH. If it is not there, it just makes software harder to use.

I would argue that too. I don't think it makes much sense for the executable to not be on the path. But that is an _installer decision_. And therefore should be fixed _in the installer_.

The Chocolatey package is following that _expected_ behaviour. 

> I basically expect both apt-get install cmake and choco install cmake

Then I would suggest you contact the maintainer of `cmake` and ask them to make adding to the PATH the default behaviour for both.

>  Especially so because I installed other dev tooling like git and make and they all were added to PATH just like I expected.

* The `git` package uses the installer, which adds it to the PATH. 
* The `make` package is simply an executable that is extracted onto the file system and Chocolatey CLI shims so it appears on the path.

The Chocolatey package does not have any special handling for either of those.

The `cmake` installer is an MSI. It is trivial to add handling in this to add `cmake.exe` to the path. That is the root problem and that's where it should be fixed.

Meanwhile, if you want it on the path add `--installargs 'ADD_CMAKE_TO_PATH=System'` to the command line (you can swap 'System' for 'User' if you want it only on the User PATH).

**Comment by nazar-pc** [comment]
Created upstream issue here: https://gitlab.kitware.com/cmake/cmake/-/issues/25795

**Comment by pauby** [comment]
Awesome. Thanks for doing that @nazar-pc.

**Comment by stenolog** [comment]
>> Wondering why the choco project closes issues that casually, degrading choco's quality:

This is a valid statement

> Please refer to our [Code of Conduct](https://github.com/chocolatey-community/.github/blob/d77df364a2449acaef47580ccb7067496042d780/CODE_OF_CONDUCT.md) before posting further.

You should read it. It's a bidirectional agreement.

And spare me the usual "own time" "community" etc. excuses.

You provide a package-manager, yet you're not able to listen(!) to users, in order to fix your processes, thus you provide similar quality than other package managers.

Is it so difficult to see what needs to happen here?

Anyway, those are rhetorical comments/questions.

Not expecting any actions.

Cu around.

**Comment by stenolog** [comment]
> --installargs 'ADD_CMAKE_TO_PATH=System'

that did not work on my side (similar to other users).

'choco' project must(!) enforce a standard-policy for standar-functionality (adding to path is absolute basic standard fucntionality, opt-out should be enforced).

**Comment by nazar-pc** [comment]
Upstream issue was closed as duplicate of https://gitlab.kitware.com/cmake/cmake/-/issues/21465, maintainer there basically says that if Chocolatey users expect `cmake` to be in Path by default, then that it is on Chocolatey to do that :confused:

**Comment by stenolog** [comment]
>> @AdmiringWorm, issues do not disappear by writing some comments, closing and locking issues.

this is not off-topic. Ask the hundred of users that hit on that issue if it is off-topic.

what a joke.

**Comment by pauby** [comment]
This issue isn't a Chocolatey package issue. This is a Windows PATH issue. That's a pretty poor response.

**Comment by TheCakeIsNaOH** [comment]
@nazar-pc @stenolog try the `cmake.portable` package, which does shim the executable to the path.

**Comment by nazar-pc** [comment]
Frankly speaking, I don't think I know or care about it enough to invest time. I only use Windows when debugging Windows-specific issues or packaging something for users.

But I do appreciate that Chocolatey exists, it made my life much easier installing various things on dev machines, thank you!

**Comment by pauby** [comment]
@nazar-pc it's my finish time (actually it was that over an hour ago) but I'll create an account over there tomorrow and see if I can lend any weight to it as I don't agree with some of what is being said over there. It doesn't make a lot of sense.

**Comment by abebeos** [comment]
> try the `cmake.portable` package

Problem is that the next user will run into the same issue.

> This issue isn't a Chocolatey package issue. This is a Windows PATH issue. That's a pretty poor response.

If you really believe this, then you should stop responding on a user-facing issue-tracker.

Surely there are other (chocolatey) team-members with better product-management/user-interaction skills.

**Comment by pauby** [comment]
> If you really believe this, then you should stop responding on a user-facing issue-tracker.
> Surely there are other (chocolatey) team-members with better product-management/user-interaction skills.

This isn't my _opinion_. This is what we have said for a very long time. Packages should follow expected behaviour of the software.

This isn't something we are going to keep debating. I appreciate you're unlikely to like that, but that is where we stand.

**Comment by abebeos** [comment]
> Your weight would be better added to the [issue linked to the cmake project, above](https://github.com/chocolatey-community/chocolatey-packages/issues/2442#issuecomment-2007913460).

Nah, I adressed the relevant recipient(s) here.

## Test Case

- new user
  - installs chocolatey
  - installs cmake via `choco install cmake`
  - runs `cmake`

This very simple test-case fails(!).

Failing tests need to be fixed.

A package-manager that does not stick to this very basic requirement has a low quality.

My personal assessment, that I'm hopefully free to express here.

**Comment by nazar-pc** [comment]
I personally didn't actually look at any descriptions, I just assumed that most of the packages are available and blindly ran `choco install cmake` hoping for the best :see_no_evil: And it worked :slightly_smiling_face:

**Comment by pauby** [comment]
If we made it clearer that the installer does not add the tool to the PATH and to try `cmake.portable`, when installing the package, it would help others in future who do this too.

**Comment by abebeos** [comment]
> This comment has been minimized. 

https://github.com/chocolatey-community/chocolatey-packages/issues/2442#issuecomment-2007960248

The comment is valid, 

Reused here:

https://gitlab.kitware.com/cmake/cmake/-/issues/21465#note_1498139

(It's getting fun, both projects are wrong, solution is trivial, but will be censored anyway, so, keeping it for me and awaiting...)

**Comment by pauby** [comment]
I added a comment to the [cmake GitLab repository](https://gitlab.kitware.com/cmake/cmake/-/issues/21465#note_1498227) which I hope at least helps. In the meantime, I raised a [PR](https://github.com/chocolatey-community/chocolatey-packages/pull/2444) to clarify the text.

As this is a heated topic, we will continue to remove or mark comments accordingly.

**Comment by pauby** [comment]
A comment was left over on the cmake GitLab site regarding [updating the MSI in future](https://gitlab.kitware.com/cmake/cmake/-/issues/21465#note_1498275).

Now that the [PR](https://github.com/chocolatey-community/chocolatey-packages/pull/2444) has been merged to clarify the text around the PATH, I'm going to go ahead and close this. I'm also going to lock it due to the heated nature of the conversation.

TL;DR for people finding this in the future, the `cmake.install` Chocolatey package follows the behaviour of the installer, which [currently](https://gitlab.kitware.com/cmake/cmake/-/issues/21465#note_1498275) does not add the tool to the PATH. If you want to have the tool added to the path, see the package page for [`cmake`](https://community.chocolatey.org/packages/cmake) or [`cmake.install`](https://community.chocolatey.org/packages/cmake.install), or install the [`cmake.portable`](https://community.chocolatey.org/packages/cmake.portable) package. Any future issues raised about this will be marked as a duplicate of this one and closed.
